src/main/java/io/supertokens/multitenancy/Multitenancy.java

```diff
@@ -190,6 +191,10 @@ public static boolean addNewOrUpdateAppOrTenant(Main main, TenantConfig newTenan
             StorageQueryException, FeatureNotEnabledException, IOException, InvalidConfigException,
             InvalidProviderConfigException, TenantOrAppNotFoundException {
 
+        if (StorageLayer.getBaseStorage(main).getType() != STORAGE_TYPE.SQL) {
```

Comment:
check if it's base tenant, and then return true, else throw unsupported exception